java/src/main/java/ai/rapids/cudf/GroupByOptions.java

```diff
@@ -52,6 +73,50 @@ public Builder withIgnoreNullKeys(boolean ignoreNullKeys) {
       return this;
     }
 
+    /**
+     * Indicates whether rows in `keys` are already sorted.
+     * The default value is false.
+     *
+     * If the `keys` are already sorted, better performance may be achieved by
+     * passing `keysSorted == true` and indicating the ascending/descending
+     * order of each column and null order by calling `withKeysDescending` and
+     * `withKeysNullSmallest`, respectively.
+     */
+    public Builder withKeysSorted(boolean keysSorted) {
+      this.keysSorted = keysSorted;
+      return this;
+    }
+
+    /**
+     * If `keysSorted == true`, indicates whether each
+     * column is ascending/descending. If empty or null, assumes all columns are
+     * ascending. Ignored if `keysSorted == false`.
+     */
+    public Builder withKeysDescending(boolean... keysDescending) {
+      if (keysDescending == null) {
+        // Use empty array instead of null
+        this.keysDescending = new boolean[0];
+      } else {
+        this.keysDescending = keysDescending;
+      }
+      return this;
+    }
+
+    /**
+     * If `keysSorted == true`, indicates the ordering
+     * of null values in each column. If empty or null, assumes all columns
+     * use 'null smallest'. Ignored if `keysSorted == false`.
+     */
+    public Builder withKeysNullSmallest(boolean... keysNullSmallest) {
```

Comment:
Don't we need/want to plumb these through for the other groupby operations?

Reply:
Before this PR, there are 3 APIs in `AggregateOperation `.  
- `aggregate()` is using the native `groupby`, so I updated to enable the new options for it.
- The other two APIs are windowing related and using the native `rolling` instead of `groupby`. So the new options can not be applied.